Sources/MessagingInApp/State/InAppMessageMiddleware.swift

```diff
@@ -0,0 +1,247 @@
+    }
+}
+
+func messageMetricsMiddleware(logger: CioInternalCommon.Logger) -> InAppMessageMiddleware {
+    middleware { _, getState, next, action in
+        let state = getState()
+        switch action {
+        case .displayMessage(let message):
+            let properties = message.gistProperties
+            // Log message view only if message is not persistent
+            if properties.persistent != true {
+                logger.debug("[InApp] Message shown, logging view for message: \(message.describeForLogs)")
+                logMessageView(logger: logger, state: state, message: message)
+            } else {
+                logger.debug("[InApp] Persistent message shown, not logging view for message: \(message.describeForLogs)")
+            }
+
+        case .dismissMessage(let message, let shouldLog, let viaCloseAction):
+            guard shouldLog else { return }
+
+            let properties = message.gistProperties
+            // Log message close only if message was dismissed via close action
+            if viaCloseAction {
+                if properties.persistent == true {
+                    logger.debug("[InApp] Persistent message dismissed, logging view for message: \(message.describeForLogs)")
+                } else {
+                    logger.debug("[InApp] Dismissed message, logging view for message: \(message.describeForLogs)")
+                }
+                logMessageView(logger: logger, state: state, message: message)
+            } else {
+                logger.debug("[InApp] Message dismissed without close action, not logging view for message: \(message.describeForLogs)")
+            }
+
+        default:
+            break
+        }
+        return next(action)
+    }
+}
+
+func messageQueueProcessorMiddleware(logger: CioInternalCommon.Logger) -> InAppMessageMiddleware {
+    middleware { dispatch, getState, next, action in
+        // Continue to next middleware if action is not processMessageQueue or if messages are empty
+        guard case .processMessageQueue(let messages) = action, !messages.isEmpty else {
+            return next(action)
+        }
+
+        let state = getState()
+        // Filter out messages with valid queueId that have not been shown yet and sort by priority
+        let notShownMessages = messages
+            .filter { message in
+                if let queueId = message.queueId {
+                    return !state.shownMessageQueueIds.contains(queueId)
+                }
+                return false
+            }
+            .reduce(into: [Message]()) { result, message in
+                if !result.contains(where: { $0.queueId == message.queueId }) {
+                    result.append(message)
+                }
+            }
+            .sorted {
+                guard let priority1 = $0.priority, let priority2 = $1.priority else {
+                    return $0.priority != nil
+                }
+                return priority1 < priority2
+            }
```

Comment:
```suggestion
.sorted { ($0.priority ?? Int.max) < ($1.priority ?? Int.max) }
```
can be simplified like this?

Reply:
Done